core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -406,6 +406,15 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
   def getBoolean(key: String, defaultValue: Boolean): Boolean = {
     getOption(key).map(_.toBoolean).getOrElse(defaultValue)
   }
+  /** Get a parameter as a Option[Boolean] */
+  def getOptionBoolean(key: String): Option[Boolean] = {
```

Comment:
I was unsure about copying the defaults from the mesos protos(they were set to None earlier which mesos translates as defaults), and `FrameworkInfo.setCheckpoint()` expects an Option[Boolean] and there was no helper function that extracted Option[Boolean] from SparkConf